### ISSUE TITLE
Image manager/picker/uploading breaks after uploading an image with very small height or width

### ISSUE THREAD

**Issue by haplo** [wagtail/wagtail#5878]
## Issue Summary

Uploading an image with very small height or width breaks the image manager and the image picker, making it impossible for users to manage images (including deleting the bad image) or uploading/choosing new images until the bad image is manually deleted from DB (either from the Django admin if it's enabled, or a Python or DB shell otherwise).

### Steps to Reproduce

1. Upload [this bad image](https://user-images.githubusercontent.com/71658/76305516-307c1280-62bd-11ea-8ef3-1841ed894ddf.png) to Wagtail.
2. */admin/images/* will then fail with an internal error with the following stacktrace:

```
Template error:
In template /home/user/Code/virtualenvs/outdoorsy-cms/lib/python3.7/site-packages/wagtail/images/templates/wagtailimages/images/results_image.html, error at line 18
   height and width must be > 0
   8 :  This behaviour caused a confusing error on the images listing view where it
   9 :  would go blank if one of the images was invalid.
   10 : 
   11 :  Separating the image rendering code into this file allows us to limit Django's
   12 :  crash/blanking behaviour to a single image so the listing can still be used when
   13 :  the issue occurs.
   14 : {% endcomment %}
   15 : 
   16 : {% load wagtailimages_tags %}
   17 : 
   18 : <div class="image"> {% image image max-165x165 class="show-transparency" alt="" %} </div>
   19 : 

Traceback:
File "virtualenv/lib/python3.7/site-packages/wagtail/images/models.py" in get_rendition
  285.                 focal_point_key=cache_key,
File "virtualenv/lib/python3.7/site-packages/django/db/models/manager.py" in manager_method
  82.                 return getattr(self.get_queryset(), name)(*args, **kwargs)
File "virtualenv/lib/python3.7/site-packages/django/db/models/query.py" in get
  408.                 self.model._meta.object_name

During handling of the above exception (Rendition matching query does not exist.), another exception occurred:
File "virtualenv/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)
File "virtualenv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)
File "virtualenv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "virtualenv/lib/python3.7/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  44.         response = view_func(request, *args, **kwargs)
File "virtualenv/lib/python3.7/site-packages/wagtail/admin/urls/__init__.py" in wrapper
  108.             return view_func(request, *args, **kwargs)
File "virtualenv/lib/python3.7/site-packages/wagtail/admin/auth.py" in decorated_view
  166.                     return view_func(request, *args, **kwargs)
File "virtualenv/lib/python3.7/site-packages/wagtail/admin/auth.py" in wrapped_view_func
  59.                 return view_func(request, *args, **kwargs)
File "virtualenv/lib/python3.7/site-packages/django/views/decorators/vary.py" in inner_func
  20.             response = func(*args, **kwargs)
File "virtualenv/lib/python3.7/site-packages/wagtail/images/views/images.py" in index
  99.             'user_can_add': permission_policy.user_has_permission(request.user, 'add'),
File "virtualenv/lib/python3.7/site-packages/django/shortcuts.py" in render
  36.     content = loader.render_to_string(template_name, context, request, using=using)
File "virtualenv/lib/python3.7/site-packages/django/template/loader.py" in render_to_string
  62.     return template.render(context, request)
File "virtualenv/lib/python3.7/site-packages/django/template/backends/django.py" in render
  61.             return self.template.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  171.                     return self._render(context)
File "virtualenv/lib/python3.7/site-packages/django/test/utils.py" in instrumented_test_render
  96.     return self.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  150.             return compiled_parent._render(context)
File "virtualenv/lib/python3.7/site-packages/django/test/utils.py" in instrumented_test_render
  96.     return self.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  150.             return compiled_parent._render(context)
File "virtualenv/lib/python3.7/site-packages/django/test/utils.py" in instrumented_test_render
  96.     return self.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  150.             return compiled_parent._render(context)
File "virtualenv/lib/python3.7/site-packages/django/test/utils.py" in instrumented_test_render
  96.     return self.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  150.             return compiled_parent._render(context)
File "virtualenv/lib/python3.7/site-packages/django/test/utils.py" in instrumented_test_render
  96.     return self.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  150.             return compiled_parent._render(context)
File "virtualenv/lib/python3.7/site-packages/django/test/utils.py" in instrumented_test_render
  96.     return self.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  62.                 result = block.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  62.                 result = block.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  188.             return template.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  173.                 return self._render(context)
File "virtualenv/lib/python3.7/site-packages/django/test/utils.py" in instrumented_test_render
  96.     return self.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/defaulttags.py" in render
  309.                 return nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/defaulttags.py" in render
  209.                     nodelist.append(node.render_annotated(context))
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/loader_tags.py" in render
  188.             return template.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  173.                 return self._render(context)
File "virtualenv/lib/python3.7/site-packages/django/test/utils.py" in instrumented_test_render
  96.     return self.nodelist.render(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render
  937.                 bit = node.render_annotated(context)
File "virtualenv/lib/python3.7/site-packages/django/template/base.py" in render_annotated
  904.             return self.render(context)
File "virtualenv/lib/python3.7/site-packages/wagtail/images/templatetags/wagtailimages_tags.py" in render
  105.         rendition = get_rendition_or_not_found(image, self.filter)
File "virtualenv/lib/python3.7/site-packages/wagtail/images/shortcuts.py" in get_rendition_or_not_found
  13.         return image.get_rendition(specs)
File "virtualenv/lib/python3.7/site-packages/wagtail/images/models.py" in get_rendition
  289.             generated_image = filter.run(self, BytesIO())
File "virtualenv/lib/python3.7/site-packages/wagtail/images/models.py" in run
  400.                 willow = operation.run(willow, image, env) or willow
File "virtualenv/lib/python3.7/site-packages/wagtail/images/image_operations.py" in run
  185.         return willow.resize((width, height))
File "virtualenv/lib/python3.7/site-packages/willow/plugins/pillow.py" in resize
  66.         return PillowImage(image.resize(size, _PIL_Image().ANTIALIAS))
File "virtualenv/lib/python3.7/site-packages/PIL/Image.py" in resize
  1806.         return self._new(self.im.resize(size, resample, box))

Exception Type: ValueError at /admin/images/
Exception Value: height and width must be > 0
```

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: no

### Technical details

* Python version: 3.7.3
* Django version: 2.2.8
* Wagtail version: 2.7 and 2.8
* Browser version: Firefox ESR 68, but it happens to users with Chrome, Edge and Safari. Not browser related.

**Comment by haplo** [comment]
I think the problem is that the rendition calculator should never set a dimension lower than 1. I will try to submit a fix for this.

**Comment by gasman** [comment]
Fixed in #5881
